File: solutions/_958_check_completeness_of_a_binary_tree/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isCompleteTree = function (root) {
  if (!root) {
    return true;
  }

  const q = [root];

  while (q.length > 0) {
    const node = q.shift();
    if (node == null) break;
    q.push(node.left);
    q.push(node.right);
  }

  return q.every((node) => node === null);
};

module.exports = { isCompleteTree, TreeNode };

/*
https://leetcode.com/problems/check-completeness-of-a-binary-tree/
*/

File: solutions/_958_check_completeness_of_a_binary_tree/solution.py
from typing import Optional
from collections import deque


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        q = deque()
        q.append(root)
        while q:
            node = q.popleft()
            if not node:
                break
            q.append(node.left)
            q.append(node.right)

        return not any(q)


# https://leetcode.com/problems/check-completeness-of-a-binary-tree/
